AEPIdentity/Sources/MobileIdentities.swift

```diff
@@ -115,6 +123,50 @@ struct MobileIdentities: Codable {
         return visitorIds
     }
 
+    /// Gets all the required identity from the Analytics extension
+    /// - Parameters:
+    ///   - event: the `Event` generated by the GetSdkIdentities API
+    ///   - sharedStateProvider: a function that can resolve `SharedState` given an extension name
+    /// - Returns: a list of all the Analytics extension identities in the form of a `UserID`
+    private func getAnalyticsIdentifiers(event: Event, sharedStateProvider: SharedStateProvider) -> [UserID] {
+        guard let analyticsSharedState = sharedStateProvider(IdentityConstants.SharedStateKeys.ANALYTICS, event) else { return [] }
+
+        var analyticsIds = [UserID]()
+
+        if let aid = analyticsSharedState.value?[IdentityConstants.Analytics.ANALYTICS_ID] as? String {
+            analyticsIds.append(UserID(namespace: MobileIdentities.NAMESPACE_ANALYTICS_AID, value: aid, type: MobileIdentities.INTEGRATION_CODE))
+        }
+
+        if let vid = analyticsSharedState.value?[IdentityConstants.Analytics.VISITOR_IDENTIFIER] as? String {
+            analyticsIds.append(UserID(namespace: MobileIdentities.NAMESPACE_USERIDENTIFIER, value: vid, type: MobileIdentities.ANALYTICS))
+        }
+
+        return analyticsIds
+
+    }
+
+    /// Gets all the required identity from the Audience extension
+    /// - Parameters:
+    ///   - event: the `Event` generated by the GetSdkIdentities API
+    ///   - sharedStateProvider: a function that can resolve `SharedState` given an extension name
+    /// - Returns: a list of all the Audience extension identities in the form of a `UserID`
+    private func getAudienceIdentifiers(event: Event, sharedStateProvider: SharedStateProvider) -> [UserID] {
+        guard let audienceSharedState = sharedStateProvider(IdentityConstants.SharedStateKeys.AUDIENCE, event) else { return [] }
+
+        var audienceIds = [UserID]()
+
+        if let dpuuid = audienceSharedState.value?[IdentityConstants.Audience.DPUUID] as? String {
+            let dpid = audienceSharedState.value?[IdentityConstants.Audience.DPID] as? String ?? ""
```

Comment:
Is it valid to have an empty dpid or should we unwrap it like the dpuuid above?

Reply:
I think it can be empty as it is data provider id.